### PR TITLE
Adds a UDPClient

### PR DESCRIPTION
**Pull request by ThadHouse**
Send only currently. Will eventually implement receive, but not
necessary for now.

### CODE REVIEW

File: src/main/native/include/udpsockets/UDPClient.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+class Logger;
+
+class UDPClient {
+  int m_lsd;
+  std::string m_address;
+  Logger& m_logger;
+
+public:
+  UDPClient(Logger& logger);
```
Comment (PeterJohnson) [inline src/main/native/include/udpsockets/UDPClient.h]: Needs to be explicit

File: src/main/native/cpp/udpsockets/UDPClient.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+#endif
+
+#include "support/Logger.h"
+#include "tcpsockets/SocketError.h"
+
+
+using namespace wpi;
+
+UDPClient::UDPClient(Logger& logger)
+    : UDPClient("", logger) { }
```
Comment (PeterJohnson) [inline src/main/native/cpp/udpsockets/UDPClient.cpp]: Extra space in `{ }`

File: src/main/native/cpp/udpsockets/UDPClient.cpp [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  }
+  addr.sin_port = htons(0);
+
+#ifdef _WIN32
+  int optval = 1;
+  setsockopt(m_lsd, SOL_SOCKET, SO_EXCLUSIVEADDRUSE, (char*)&optval, sizeof optval);
+#else
+  int optval = 1;
+  setsockopt(m_lsd, SOL_SOCKET, SO_REUSEADDR, (char*)&optval, sizeof optval);
+#endif
```
Comment (PeterJohnson) [inline src/main/native/cpp/udpsockets/UDPClient.cpp]: Neither one of these should be necessary since you're not binding an actual port..

**Review by PeterJohnson — CHANGES_REQUESTED** [review]

File: src/main/native/cpp/udpsockets/UDPClient.cpp (hunk 3 above)
Comment (ThadHouse) [reply src/main/native/cpp/udpsockets/UDPClient.cpp]: Fixed here. My PR later to add in receive will add this back in in a modified form.

File: src/main/native/include/udpsockets/UDPClient.h (hunk 1 above)
Comment (ThadHouse) [reply src/main/native/include/udpsockets/UDPClient.h]: Fixed

File: src/main/native/cpp/udpsockets/UDPClient.cpp (hunk 2 above)
Comment (ThadHouse) [reply src/main/native/cpp/udpsockets/UDPClient.cpp]: Fixed

**Review by PeterJohnson — APPROVED** [review]

**Pull request merged**